sammi/cdf_attribute_manager.py

```diff
@@ -197,24 +356,91 @@ def get_global_attributes(self, instrument_id: str | None = None) -> dict:
             elif attr_schema["required"] and attr_name not in self._global_attributes:
                 # TODO throw an error
                 output[attr_name] = None
-
         return output
 
-    def load_variable_attributes(self, file_name: str) -> None:
+    def global_attribute_template(self) -> OrderedDict:
         """
-        Add variable attributes for a given filename.
+        Function to generate a template of required global attributes
+        that must be set for a valid data file.
+
+        Returns
+        -------
+        template : `OrderedDict`
+            A template for required global attributes that must be provided.
+        """
+        template = OrderedDict()
+        for attr_name, attr_schema in self.global_attribute_schema.items():
+            if attr_schema["required"] and attr_name not in self._global_attributes:
+                template[attr_name] = None
+        return template
+
+    def global_attribute_info(self, attribute_name: Optional[str] = None):
```

Comment:
I don't love that this requires pandas... although I'm ok with importing/depending on pandas in general, I personally don't think this function justifies it. Could we instead return a python dictionary that can easily be imported into pandas using `from_dict`? I.e. just update the schema to include only `attribute_name` and return that?

Reply:
Also, this function is missing a return typehint

Reply:
Good call - I'll add the typehint. 

In terms of the dependency its fully optional for the package, only needed for this function.

Reply:
I added typehints for these functions. I want to keep this as a `DataFrame` since I use it in the documentation to auto-generate CSV tables in the CDF Format Guide. I want to leave it as optional though since I know HERMES will overwrite these functions to use `astropy.Table` instead. If you feel strongly I can update the documentation code to create CSVs from the dicts instead of dataframes.

Reply:
If it's not too much work, I would prefer to remove the Pandas dependency here. I don't have super strong feelings about it only because I suspect that we will end up importing Pandas at some point... but it is a pretty large dependency, and we don't really need it right now.